### PR TITLE
Fix #376

### LINKED ISSUE #376: RepeatedFunc wrongly throws exception
I expect the following code to run without a problem :
```java
        new RepeatedFunc<>(
            input -> {return null;}, 2
        ).apply(true);
```
Instead I got an exception : `java.lang.IllegalArgumentException: Repeat counter is equal or less than zero: 2`
I don't expect `RepeatedFunc` to behave like `NoNull` implementations, and even if it is the case, the exception message is completely wrong.
The implementation of `RepeatedFunc.apply` should be fixed to check for invalid `this.times` values instead of a `null` result.

### PR DESCRIPTION
**Pull request by ixmanuel**
```java
// IllegalStateException
// The result <null> is an invalid value in Cactoos:
//     - If required, try a null object instead.
new RepeatedFunc<>(
    input -> {return null;}, 2
).apply(true);

// IndexOutOfBoundsException
// Unexpected value=0:
//     - Expected a value > 0 
new RepeatedFunc<>(
    input -> {return 1;}, 0
).apply(true);

```

### CODE REVIEW

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#381` is in scope.

**Comment by 0crat** [comment]
+15 points just awarded to @ixmanuel, total is [+75](http://www.0crat.com/u/ixmanuel).

**Comment by yegor256** [comment]
@ixmanuel it looks like over-design. First, we already have `Scalar.NoNulls`. Second, I have big doubts about this new `ValidBound` class -- it looks very artificial and not really Java-ish. Third, the problem you're fixing can be fixed with just 2-3 lines of code.

**Comment by ixmanuel** [comment]
@yegor256, now it is in kiss-mode. Thanks for your feedback in order to be in the same channel.

Some thoughts:
1. Null
```java
// Func (not used because it rises an IO exception)
return new Func.NoNulls<>(this.func).apply(input);

// Scalar (used with an additional statement and new scalar ()->rslt)
final Y rslt = result;
return new Scalar.NoNulls<Y>(()->rslt).value();

// Proposed (deprecated)
return new NoNull(result).value()
```
2. ValidBound's intent is to have uniform-message-exceptions-handling for at least, at most, greater than and less than cases. Not just for this particular case and yes, it is not Java-ish. I'll keep it in mind, thanks.

File: src/main/java/org/cactoos/func/RepeatedFunc.java [hunk 1]
```diff
@@ -75,19 +75,17 @@ public RepeatedFunc(final Func<X, Y> fnc, final int max) {
...
-            throw new IllegalArgumentException(
-                new FormattedText(
-                    "Repeat counter is equal or less than zero: %d",
-                    this.times
-                ).asString()
-            );
-        }
-        return result;
+        final Y rslt = result;
+        return new Scalar.NoNulls<Y>(()->rslt).value();
```
Comment (yegor256) [inline src/main/java/org/cactoos/func/RepeatedFunc.java]: @ixmanuel why do we need that? Why can't that function return `null`? I think this is something our users will complain about. This functionality is not what the class is supposed to do.

**Comment by yegor256** [comment]
@ixmanuel see above

**Comment by ixmanuel** [comment]
@yegor256 done :)

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#381` assigned to @kkamkou, please go ahead ([policy](http://datum.zerocracy.com/pages/policy.html)).

**Comment by 0crat** [comment]
@kkamkou resigned from `gh:yegor256/cactoos#381`, please stop working.

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#381` assigned to @kkamkou, please go ahead ([policy](http://datum.zerocracy.com/pages/policy.html)).

**Comment by 0crat** [comment]
Order was successfully finished: +30 points just awarded to @kkamkou, total is [+930](http://www.0crat.com/u/kkamkou).

**Pull request merged**